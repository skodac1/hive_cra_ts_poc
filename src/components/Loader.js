import React from 'react'
import PropTypes from 'prop-types'
import { Box } from 'rebass/styled-components'
import ClipLoader from 'react-spinners/ClipLoader'

const Loader = ({ size }) => {
  return (
    <Box
      sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%',
        '& span': {
          border: '3px solid',
          borderColor: 'primary',
          borderBottomColor: 'transparent',
        },
      }}
    ><ClipLoader size={size} /></Box>
  )
}

Loader.propTypes = {
  size: PropTypes.number,
}

Loader.defaultProps = {
  size: 30,
}

export default Loader
