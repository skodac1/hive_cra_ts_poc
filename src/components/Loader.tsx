import React from 'react'
import { Box } from 'rebass/styled-components'
import ClipLoader from 'react-spinners/ClipLoader'

const Loader = ({ size }: LoaderProps) => {
  return (
    <Box
      sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        height: '100%',
        '& span': {
          border: '3px solid',
          borderColor: 'primary',
          borderBottomColor: 'transparent',
        },
      }}
    ><ClipLoader size={size} /></Box>
  )
}

Loader.defaultProps = {
    size: 30,
}

type LoaderProps = { size?: number };

export default Loader
