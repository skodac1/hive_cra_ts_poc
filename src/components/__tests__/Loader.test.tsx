import * as React from 'react';
import { render } from '@testing-library/react';
import Loader from "../Loader";

describe('<Loader />', () => {
    it('matches snapshot', () => {
        const { baseElement } = render(<Loader />);
        expect(baseElement).toMatchSnapshot();
    });
})