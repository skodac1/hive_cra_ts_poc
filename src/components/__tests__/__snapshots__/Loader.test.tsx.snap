// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ WjSqt"
    >
      <span
        class="css-1yzshy8"
      />
    </div>
  </div>
</body>
`;
